### PR TITLE
Update React README

### PR DESCRIPTION
**Pull request by tquetano-r7**
Resolves:
- #71
- #72
- #73
- #74
- #75
- #76
- #77
- #78
- #79
- #80

### CODE REVIEW

**Comment by tquetano-r7** [comment]
I basically just did the entire React README issue block in one fell swoop

File: react/README.md [hunk 1]
```diff
@@ -18,35 +18,25 @@
...
-
-  // good
-  class Listing extends React.Component {
-    render() {
-      return <div />;
-    }
-  }
-  ```
+    + Always use JSX syntax.
+    + Prefer the use of functional components when state and lifecycle methods are not needed.
```
Comment (rzec-r7) [inline react/README.md]: We should make mention that functional components don't support refs unless you are using React 16.3.0 or later which has the forwardRef API (https://reactjs.org/blog/2018/03/29/react-v-16-3.html#forwardref-api).

File: react/README.md [hunk 2]
```diff
@@ -74,56 +64,139 @@
...
   // good
   export default class ReservationCard extends Component {
+    static displayName = 'ReservationCard';
+  }
+  
+  export default function ReservationCard() {
+    //
   }
+  
+  ReservationCard.displayName = 'ReservationCard';
```
Comment (rzec-r7) [inline react/README.md]: if we are going to be using static for `propTypes`, `defaultProps`, and `contextTypes`, does it make sense to also use it for `displayName`?

File: react/README.md [hunk 3]
```diff
@@ -202,8 +290,8 @@
   // good
   <Foo className="stuff"/>
```
Comment (rzec-r7) [inline react/README.md]: update this to have a space at the end `<Foo className="stuff" />`

File: react/README.md (hunk 2 above)
Comment (tquetano-r7) [reply react/README.md]: yup, ill update

File: react/README.md (hunk 1 above)
Comment (tquetano-r7) [reply react/README.md]: yup, ill add some verbiage in there

File: react/README.md (hunk 3 above)
Comment (tquetano-r7) [reply react/README.md]: yup, good catch

**Pull request merged**